components/report-details/support/form.tsx

```diff
@@ -0,0 +1,289 @@
+		title: "Thank you! We got your support.",
+		content:
+			"Your transaction was successful. We're grateful for your contribution!",
+	},
+};
+
+async function getOrdersForReport(
+	hypercertClient: HypercertExchangeClient | null,
+	hypercertId: Partial<Report>["hypercertId"],
+	chainId?: number,
+) {
+	if (!hypercertId) {
+		console.warn("[Fetching orders] - No hypercert ID provided");
+		return [];
+	}
+	if (!hypercertClient) {
+		console.warn("[Fetching orders] - No hypercert client found");
+		return [];
+	}
+	if (chainId === undefined) {
+		console.warn("[Fetching orders] - No chainId provided");
+		return [];
+	}
+
+	try {
+		const { data: orders } = await hypercertClient.api.fetchOrdersByHypercertId(
+			{
+				hypercertId,
+				chainId,
+			},
+		);
+		return orders;
+	} catch (error) {
+		console.error("[Fetching orders] - Error fetching orders", error);
+		return [];
+	}
+}
+
+const SupportReportForm = ({
+	drawerContainer,
+	hypercertId,
+}: SupportReportFormProps) => {
+	const { address, isConnected, chainId } = useAccount();
+	const provider = useEthersProvider({ chainId });
+	const signer = useEthersSigner({ chainId });
+	const publicClient = usePublicClient({ chainId });
+	const { dollarAmountNeeded, pricePerUnit } = useFunding();
+
+	const HCExchangeClient = new HypercertExchangeClient(
+		chainId ?? sepolia.id,
+		// @ts-ignore
+		provider,
+		signer,
+	);
+
+	const { handleBuyFraction, transactionStatus, transactionHash } =
+		useHandleBuyFraction(publicClient, HCExchangeClient);
+
+	const {
+		isPending: isOrdersPending,
+		error: orderError,
+		data: orders,
+	} = useQuery({
+		queryKey: ["ordersFromHypercert"],
+		queryFn: () =>
+			getOrdersForReport(
+				HCExchangeClient,
+				// TODO: Replace with actual hypercert ID
+				"0xa16dfb32eb140a6f3f2ac68f41dad8c7e83c4941-39472754562828861761751454462085112528896",
+				// "0xa16dfb32eb140a6f3f2ac68f41dad8c7e83c4941-67375908650345815765748172271490105868288",
+				chainId,
+			),
+	});
+
+	const { form, onSubmit, isProcessing } = useSupportForm(
+		Number(dollarAmountNeeded),
+		pricePerUnit,
+		orders?.[5],
+		handleBuyFraction,
+		address,
+		hypercertId,
+	);
+
+	if (!isConnected && !address) {
+		return (
+			<div className="flex flex-col gap-4 p-4">
+				<div className="flex flex-col gap-4 justify-center items-center">
+					<h4 className="font-bold text-center">
+						Connect your wallet to support this report
+					</h4>
+					<ConnectButton />
+				</div>
+			</div>
+		);
+	}
+
+	return (
+		<section>
+			{!isProcessing && (
+				<Form {...form}>
+					<form
+						onSubmit={form.handleSubmit(onSubmit)}
+						className="flex flex-col gap-4"
+					>
+						<FormField
+							control={form.control}
+							name="fractionPayment"
+							render={({ field }) => (
+								<FormItem>
+									<FormLabel>Support amount</FormLabel>
+									<div className="grid grid-cols-5 gap-4">
+										{[5, 10, 20, 50, 100].map((amount) => (
+											<button
```

Comment:
Why arn't you using the UI component Button
```suggestion
											<Button
```